main.py

```diff
@@ -481,6 +484,92 @@ def delete_interface_metadata(self, request: Request) -> JSONResponse:
         self.notify_metadata_changes(interface, 'removed')
         return JSONResponse("Operation successful")
 
+    @staticmethod
+    def _get_tag_type(content):
+        tag_type = content.get("tag_type", '1')
+        if tag_type != str(TAGType.VLAN.value):
+            detail = f"The TAG type {tag_type} is not allowed."
+            raise HTTPException(400, detail=detail)
+        return tag_type
+
+    @staticmethod
+    def check_non_range(tag_range):
+        """Change integer or singular interger list to
+        list[int, int] when necessary"""
+        if isinstance(tag_range, int):
+            tag_range = [tag_range] * 2
+        elif len(tag_range) == 1:
+            tag_range = [tag_range[0]] * 2
+        return tag_range
+
+    def _get_tag_ranges(self, request: Request):
+        """Get tag_ranges and check validity:
+        - It should be ordered
+        - Not unnecessary partition (eg. [[10,20],[20,30]])
+        - Singular intergers are changed to ranges (eg. [10] to [[10, 10]])
+        The ranges are understood as [inclusive, inclusive]"""
+        content_type_json_or_415(request)
+        content = get_json_or_400(request, self.controller.loop)
+        ranges: list[Union[int, list[int]]] = content["tag_ranges"]
+        tag_type = self._get_tag_type(content)
+        ranges[0] = self.check_non_range(ranges[0])
```

Comment:
You're accessing the first index of the range without checking if it's not empty, and the spec validation isn't validating a length yet, so looks like this can lead to IndexError if you can double check this.

Reply:
Added if statement to avoid Index errors, also openapi has a restriction for empy lists

Reply:
Nice